lib/elixir/lib/module.ex

```diff
@@ -861,6 +861,33 @@ defmodule Module do
     end, tuples)
   end
 
+  @spec make_overridable(module, module) :: :ok
+  def make_overridable(module, behaviour) when is_atom(module) and is_atom(behaviour) do
```

Comment:
What if the given module is not a behaviour? We need to check it and raise accordingly. I think we need to do something like this:

```
unless Code.ensure_compiled?(module) and function_exported?(module, :behaviour_info, 1) do
  raise "oops"
end
```